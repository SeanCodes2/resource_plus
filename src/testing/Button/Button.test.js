import React from "react";

import { createRoot } from 'react-dom/client'
import Button from "./Button";

//the below items are imported to bring in specific test functions. first are some tools that allow for shallow rendering tests and cleanup of data after each test. second are tools actually used to make the tests function
import {screen, render, cleanup} from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
//npm install react-test-renderer  --  gives us rendering functionality for tests
import renderer from 'react-test-renderer'
import ShallowRender from '../ShallowRender'
import EndToEnd from '../EndToEnd'

//the below function will clean up data after each test is run
afterEach(cleanup)

//each test below will begin with it('string name of test', function to perform)
it('Button renders without crashing', () => {
    const div = document.createElement('div')
    const root = createRoot(div)
    root.render(<Button />)
})

it('Button renders prop correctly', () => {
    const {getByTestId} = render(<Button label='Submit' />)
    expect(screen.getByTestId('button')).toHaveTextContent('Submit')
})

it('Button matches snapshot', () => {
    const tree = renderer.create(<Button label='Save' />).toJSON()

    //When this test runs, it creates a tree taht will cause a folder called __snapshots__ for a file called Button.test.js.snap
    //It will save what renders in that snapshot file and each snapshot to see the differences.
    expect(tree).toMatchSnapshot()
})

it('End to End renders props correctly', () => {
    const {getByTestId} = render(<EndToEnd test='test' />)
    expect(screen.getByTestId('paragraph1')).toHaveTextContent('test')
})

it('ShallowRender renders correctly', () => {
    const tree = renderer.create(<ShallowRender />).toJSON()

    expect(tree).toMatchSnapshot()
})