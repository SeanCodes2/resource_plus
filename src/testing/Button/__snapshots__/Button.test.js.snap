// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button matches snapshot 1`] = `
<div
  className="text-center"
>
  <button
    className="btn btn-warning"
    data-testid="button"
  >
    Save
  </button>
</div>
`;

exports[`ShallowRender renders correctly 1`] = `
<div
  className="col-md-6"
>
  <article
    className="border-dark border rounded m-1 p-4"
  >
    <h2
      className="text-center p-2"
    >
      Shallow Rendering
    </h2>
    <p>
      Shallow Rendering allows us to test in a simplier environment than a browser to ensure our React Component code is functioning without errors.
    </p>
    <p>
      We will create a button component in the testing components/testing folder.
    </p>
    <ol>
      Follow the steps below to create a simple test:
      <li>
        Create a Button folder in the testing folder
      </li>
      <li>
        Create a Button.js file that houses a Button component
      </li>
      <li>
        Create a Button.test.js file to write the unit tests
      </li>
      <li>
        Create the tests to render or to ensure props are displaying correctly. See Button.test.js for examples
      </li>
    </ol>
  </article>
</div>
`;
